enzyme/Enzyme/MLIR/Dialect/EnzymeOps.td

```diff
@@ -127,8 +127,17 @@ def ForwardDiffOp : Enzyme_Op<"fwddiff",
     $fn `(` $inputs `)` attr-dict `:` functional-type($inputs, results)
   }];
 
-  // Define activity optimizations as canonicalization 
   let hasCanonicalizer = 1;
+
+  let extraClassDeclaration = [{
+  ::llvm::SmallVector<::mlir::Value> getPrimalArgs() {
+    return ::mlir::enzyme::detail::filterGradInputs<ForwardDiffOp, false>(*this);
+  }
+
+  ::llvm::SmallVector<::mlir::Value> getGradArgs() {
```

Comment:
We should likely separate this is getshadow args [aka the dup ones] as distinct from the autodiff differential return